src/sentry/issues/occurrence_consumer.py

```diff
@@ -392,21 +393,40 @@ def _process_batch(worker: ThreadPoolExecutor, message: Message[ValuesBatch[Kafk
         except Exception:
             logger.exception("Failed to unpack message payload")
             continue
+
         # group by the fingerprint, there should only be one of them
         partition_key: str = payload["fingerprint"][0] if payload["fingerprint"] else ""
-        occcurrence_mapping[partition_key].append(payload)
+
+        payload_type = message.get("payload_type", PayloadType.OCCURRENCE.value)
+        match payload_type:
+            case PayloadType.STATUS_CHANGE.value:
+                status_change_mapping[partition_key].append(payload)
+            case PayloadType.OCCURRENCE.value:
+                occcurrence_mapping[partition_key].append(payload)
+            case _:
+                metrics.incr(
+                    "occurrence_consumer._process_batch.dropped_invalid_payload_type",
+                    sample_rate=1.0,
+                    tags={"payload_type": payload_type},
+                )
 
     # Number of occurrences that are being processed in this batch
     metrics.gauge("occurrence_consumer.checkin.parallel_batch_count", len(batch))
 
     # Number of groups we've collected to be processed in parallel
     metrics.gauge("occurrence_consumer.checkin.parallel_batch_groups", len(occcurrence_mapping))
-    # Submit occurrences for processing
+    # Submit occurrences & status changes for processing
     with sentry_sdk.start_transaction(op="process_batch", name="occurrence.occurrence_consumer"):
-        futures = [
+        occurrence_futures = [
             worker.submit(process_occurrence_group, group) for group in occcurrence_mapping.values()
         ]
-        wait(futures)
+        status_change_futures = [
+            worker.submit(process_occurrence_group, group[-1])
+            for group in status_change_mapping.values()
+        ]
+
+        wait(occurrence_futures)
+        wait(status_change_futures)
```

Comment:
Oh sorry, what I meant was to do this within `process_occurrence_group`. If we split them all up here, we have no guarantee that `status_changes` related to a specific occurrence will be processed after.

I was just thinking that within an occurrence group, maybe it makes sense to split them up? I'm not sure if there's ever a risk of a status update somehow ending up on the topic before an occurrence, but maybe that just doesn't make sense

Reply:
Hmm... OK, let me move that logic...